bridge-token-factory/tests/event_serialization.rs

```diff
@@ -34,18 +35,47 @@ fn real_data_locked_event_serialization() {
 }
 
 fn rand_string(rng: &mut ThreadRng) -> String {
-    (0..rng.gen::<u8>())
-        .map(|_| (0x20u8 + (rng.gen::<f32>() * 96.0) as u8) as char)
-        .collect()
+    const CHARSET: &[u8] = b"abcdefghijklmnopqrstuvwxyz0123456789._-";
+    let string_len = rng.gen_range(2, 64);
+
+    'main: loop {
+        let rand_str: String = (0..string_len)
+            .map(|_| {
+                let idx = rng.gen_range(0, CHARSET.len());
+                CHARSET[idx] as char
+            })
+            .collect();
+
+        let mut last_char_is_separator = true;
+
+        for c in rand_str.chars() {
+            let current_char_is_separator = match c {
+                '-' | '_' | '.' => true,
+                _ => false,
+            };
+
+            if current_char_is_separator && last_char_is_separator {
+                continue 'main;
+            }
+
+            last_char_is_separator = current_char_is_separator;
+        }
+
+        if !last_char_is_separator {
+            return rand_str;
+        }
+    }
```

Comment:
Because it is more complicated rule for account_id: 
* Only 'a'-'z', '0'-'9', '-', '_', '.' chars
* No two separators in a row
* No separator in the end

Reply:
Please check my related comment below (couldn't put it here as I wanted to suggest some code)